tests/test_views.py

```diff
@@ -0,0 +1,130 @@
+
+from blog.models import Post
+
+
+class ViewsTest(TestCase):
+    """Testing class for views."""
+
+    USERNAME = 'admin'
+    PASSWORD = '12345678'
+
+    def setUp(self):
+        """Prepare data for testing."""
+        self.client = Client()
+        self.user = User.objects.create(
+            username='testuser',
+            password='blablabla',
+            is_superuser=True, is_staff=True,
+            email='testuser@gmail.com',
+            is_active=True,
+        )
+
+    def test_index_view_rendering(self):
+        """Testing main page for render needed posts."""
+        time_zone = timezone.get_current_timezone()
+        post = Post.objects.create(
+            author=self.user, title='Test',
+            text='superText',
+            created_date=datetime(
+                day=1, month=3,
+                year=2020, tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1, month=3,
+                year=2020,
+                tzinfo=time_zone,
+            ),
+        )
+        past_post = Post.objects.create(
+            author=self.user, title='past_est',
+            text='superText',
+            created_date=datetime(
+                day=1, month=4,
+                year=2019,
+                tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1, month=4,
+                year=2019,
+                tzinfo=time_zone,
+            ),
+        )
+        future_post = Post.objects.create(
+            author=self.user,
+            title='future_test',
+            text='superText',
+            created_date=datetime(
+                day=1, month=4,
+                year=2120,
+                tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1,
+                month=4,
+                year=2120,
+                tzinfo=time_zone,
+            ),
+        )
+        with patch(
+                'django.utils.timezone.now', lambda: datetime(
+                    day=1,
+                    month=1,
+                    year=2020,
+                    tzinfo=time_zone,
+                ),
+        ):
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(list(response.context['posts']), [past_post])
+            self.assertNotContains(response, post)
+            self.assertNotContains(response, future_post)
+        with patch(
+                'django.utils.timezone.now', lambda: datetime(
+                    day=1,
+                    month=4,
+                    year=2020,
+                    tzinfo=time_zone,
+                ),
+        ):
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(
+                list(response.context['posts']), [
+                    post,
+                    past_post,
+                ],
+            )
+            self.assertNotContains(response, future_post)
```

Comment:
You should change [this](https://github.com/kpi-web-guild/django-girls-blog-igorkruglyak/blob/372a0c5/blog/views.py#L12) line to `order_by('-published_date')` to retrieve an order you are expect in this test.